.github/workflows/ci.yml

```diff
@@ -9,21 +9,109 @@ on:
-        elixir: ['1.13']
-        otp: ['24']
-        rebar3: ['3.16.1']
+        triplet:
+          - otp: '24'
+            elixir: '1.13'
+            rebar3: '3.22.1'
+          - otp: '25'
+            elixir: '1.14'
+            rebar3: '3.22.1'
+          - otp: '26'
+            elixir: '1.15'
+            rebar3: '3.22.1'
+          - otp: '27.0-rc1'
+            elixir: '1.16'
+            rebar3: '3.22.1'
     steps:
       - uses: actions/checkout@v2
       - uses: erlef/setup-beam@v1
         with:
-          elixir-version: ${{matrix.elixir}}
-          otp-version: ${{matrix.otp}}
-          rebar3-version: ${{matrix.rebar3}}
+          otp-version: ${{matrix.triplet.otp}}
+          elixir-version: ${{matrix.triplet.elixir}}
+          rebar3-version: ${{matrix.triplet.rebar3}}
       - name: Static analysis
         run: rebar3 dialyzer
       - name: Common Tests
         run: rebar3 ct
+
+  gen_ex_doc:
+    name: Generate `ex_doc` for `priv`
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      matrix:
+        triplet:
+          - otp: '24'
+            elixir: '1.13'
+            cache-id: oldest
+          - otp: '25'
+            elixir: '1.14'
+            cache-id: plus_one
+          - otp: '26'
+            elixir: '1.15'
+            cache-id: plus_two
+          - otp: '27.0-rc1'
+            elixir: '1.16'
+            cache-id: plus_three
+    steps:
+      - name: Setup Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{matrix.triplet.otp}}
+          elixir-version: ${{matrix.triplet.elixir}}
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Build ex_doc
+        run: |
+          mix deps.get
+          mix escript.build
+      - name: Save cache for consumption  # We do this to use the file in a non-Elixir env.
+        uses: actions/cache/save@v4
+        with:
+          path: priv
+          key: consumer-cache-${{matrix.triplet.cache-id}}
+
+  consumer:
+    name: "Consumer: Erlang/OTP ${{matrix.otp}} / rebar3 ${{matrix.rebar3}}"
+    runs-on: ubuntu-latest
+    needs: gen_ex_doc
+    strategy:
+      matrix:
+        otp: ['24', '25', '26', '27.0-rc1']
+        rebar3: ['3.22.1']
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Restore cache (oldest)  # We do this to use the file in a non-Elixir env.
+        uses: actions/cache/restore@v4
+        with:
+          path: priv
+          key: consumer-cache-oldest
+      - name: Restore cache (plus_one)
+        uses: actions/cache/restore@v4
+        with:
+          path: priv
+          key: consumer-cache-plus_one
+      - name: Restore cache (plus_two)
+        uses: actions/cache/restore@v4
+        with:
+          path: priv
+          key: consumer-cache-plus_two
+      - name: Restore cache (plus_three)
+        uses: actions/cache/restore@v4
+        with:
+          path: priv
+          key: consumer-cache-plus_three
```

Comment:
... and then recover them for execution. Ofc, at this particular moment the names are "fixed" to a four version window, but this might change/be tweaked.